pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -461,7 +461,11 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream ads.AggregatedDiscove
 
 			case RouteType:
 				routes := discReq.GetResourceNames()
-				if len(routes) == len(con.Routes) || len(routes) == 0 {
+
+				sort.Strings(routes)
+				sort.Strings(con.Routes)
+
+				if reflect.DeepEqual(con.Routes, routes) || len(routes) == 0 {
```

Comment:
better to do the check on `len(routes)` first if you are about to do deep equal as it may be true and save the reflection time.

Reply:
Done, PTAL :-)